### PR TITLE
[REVIEW] Use ColumnViewAccess on Host side [skip ci]

### LINKED ISSUE #6339: [FEA] Use ColumnViewAccess on Host side
**Is your feature request related to a problem? Please describe.**
Just like the device side, the host side can use the `ColumnViewAccess` interface and remove some of its current methods to be overrides of this interface.
**Describe the solution you'd like**
Nice to have for coherency and readability, implement `ColumnViewAccess` for `HostColumnVectorCore`

### PR DESCRIPTION
**Pull request by kuhushukla**
Fixes #6339
Has a lot of commits but the relevant one is f54f418 since others are from another PR in flight #6328

I have taken the liberty to rename some of the interface api methods with no impact to the plugin code to make it more consistent.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by kuhushukla** [comment]
@revans2 Could u please take a look at f54f418 commit for the relevant changes? Everything else is essentally #6328

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 - PR #6254 Add `cudf::make_dictionary_from_scalar` factory function
 - PR #6262 Add nth_element series aggregation with null handling
 - PR #6301 Add JNI bindings to nvcomp
+- PR #6328 Java and JNI bindings for getMapValue/map_lookup
```
Comment (revans2) [inline CHANGELOG.md:18]: You need your own CHANGELOG entry.

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java [hunk 2]
```diff
@@ -101,7 +101,7 @@ public DType getDataType() {
     }
 
     @Override
-    public long getNumRows() {
+    public long getRowCount() {
```
Comment (revans2) [inline java/src/main/java/ai/rapids/cudf/ColumnVector.java:104]: You cannot delete a public API like this.  We need to deprecate the old one.

Comment (kuhushukla) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java:104]: Sure, let me do that.

File: CHANGELOG.md (hunk 1 above)
Comment (kuhushukla) [reply CHANGELOG.md:18]: done.

File: java/src/main/java/ai/rapids/cudf/ColumnVector.java (hunk 2 above)
Comment (kuhushukla) [reply java/src/main/java/ai/rapids/cudf/ColumnVector.java:110]: done.

**Comment by kuhushukla** [comment]
@revans2 addressed the comments. Please take a look.

**Comment by revans2** [comment]
This looks fine for now, but I would like to look at it again once everything else is merged in that this depends on.

**Comment by kuhushukla** [comment]
> This looks fine for now, but I would like to look at it again once everything else is merged in that this depends on.

Sure. Thanks.

**Comment by kuhushukla** [comment]
@revans2 , I have upmerged since all the related PRs are in now, please take a look. Appreciate it.

**Review by revans2 — APPROVED** [review]

**Pull request merged**